### PR TITLE
Some of the Jobs Endpoints

### PR DESCRIPTION
**Pull request by andrewctam**
- Add shared state between routes
- Move routes to separate files under /routes

Implemented:
- POST route /upload
- GET route /get-history
- PUT route /remove-from-history
- PUT route /clear-history
- PUT route /add-job
- GET route /job-list
- GET route /job-info/:jobID

TODO: Still need to implement job logic to transfer the files, in another PR, likely in start-job which can be called by add-job

https://github.com/daminals/orcanet-rust/assets/54547089/d6fe4a9a-5679-4fcb-9160-c8a960e86e31

### CODE REVIEW

**Comment by andrewctam** [comment]
@daminals @ingrettable Can you check if these changes are ok before I continue with the other endpoints?

**Comment by andrewctam** [comment]
@daminals @ingrettable I think this PR is in a good state for review, before the changes become too large. We also have some changes in the "jobs-endpoints-market" branch for peer related routes, but those can be reviewed later once the Sea Horse market is complete.

**Comment by xiej2520** [comment]
Note: the `get_jobs_state` method returns a clone of the jobs state, some of the route handlers that modify it do not seem to take that account, the changes they make will not show up.
`get_jobs_state` should be replaced with `jobs() -> &Jobs` and `jobs_mut() -> &mut Jobs`, I have another commit doing that but I'll hold off until we finish merging Seahorse's market.

**Comment by xiej2520** [comment]
I branched from this onto https://github.com/daminals/orcanet-rust/tree/market_peer_merge_route,
with SeaHorses' market implementation (currently waiting on full implementation).
I restructured the endpoints to be by team and am periodically merging Sea Horses' commits onto the branch, so hopefully we can get a fully working market + peer + server this week.

**Comment by daminals** [comment]
Instead of having the jobs/files endpoints located within the producer module, it would probably make sense to make a separate server model, since the local producer is not going to be handling the jobs nor downloading files on behalf of the client, that would either be consumer (but it shouldn't go into consumer because splitting endpoints into multiple modules is a maintenance disaster) or server functionality

**Comment by andrewctam** [comment]
I moved files.rs and jobs.rs to /transfer. There is already a server.rs so I don't think I could have called it server.

File: peernode/src/peer-node/routes/file_routes.rs
Comment (daminals) [inline peernode/src/peer-node/routes/file_routes.rs]:
```suggestion
```

Remove unused comments

File: peernode/src/peer-node/routes/file_routes.rs
Comment (daminals) [inline peernode/src/peer-node/routes/file_routes.rs]:
```suggestion
    let producer = "this arg was removed";
```

**Review by daminals — CHANGES_REQUESTED** [review]

**Comment by daminals** [comment]
Have you been able to test these endpoints with the actual desktop client?

**Comment by andrewctam** [comment]
I believe @xiej2520 tried, Jacky did you have success with it?

**Comment by xiej2520** [comment]
I tried with orcanet-desktop but I don't think it has all the endpoints implemented, it has `get-peers` but we haven't done that yet (waiting on market).

**Review by daminals — APPROVED** [review]

**Pull request merged**
